Index: app/src/main/java/de/blitzkrieg/motormanager/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/de/blitzkrieg/motormanager/MainActivity.java	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/java/de/blitzkrieg/motormanager/MainActivity.java	(date 1559575174254)
@@ -1,12 +1,14 @@
 package de.blitzkrieg.motormanager;
 
 import android.annotation.SuppressLint;
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 import android.view.View;
 import android.view.WindowManager;
 import android.widget.AdapterView;
 import android.widget.ArrayAdapter;
+import android.widget.EditText;
 import android.widget.ListView;
 import android.widget.Switch;
 import android.widget.TextView;
@@ -29,9 +31,25 @@
     private TextView mKmh;
     private TextView mRpm;
 
-    public MainActivity() {
-    }
+    public static String EXTRA_MESSAGE;
+    public String msg;
+    //reaction after selection of an BT-Device to connect
+    private AdapterView.OnItemClickListener myListClickListener = new AdapterView.OnItemClickListener() {
+        public void onItemClick(AdapterView av, View v, int arg2, long arg3) {
+            // Get the device MAC address, the last 17 chars in the View
+            String info = ((TextView) v).getText().toString();
+            final String address = info.substring(info.length() - 17);
+
+            mBluetoothStatus.setText(R.string.connecting);
+            mConnect.setText(R.string.connecting);
+            mDeviceList.setVisibility(View.INVISIBLE);
+            mBTManager.connect(address);
+        }
+    };
 
+    public void setMsg(String msg) {
+        this.msg = msg;
+    }
 
     @SuppressLint("HandlerLeak")
     @Override
@@ -61,6 +79,16 @@
                 mBluetoothStatus.setText(msg);
                 mAnzParcels++;
                 mTextAnzParcels.setText(mAnzParcels + "");
+                setMsg(msg);
+                int msgZ = Integer.parseInt(msg);
+                do {
+                    if (msgZ < 300) {
+                        mKmh.setText(msg);
+                    } else {
+                        mRpm.setText(msg);
+                    }
+
+                } while (msgZ != 0);
             }
 
             @Override
@@ -106,11 +134,20 @@
         {
             public void onClick(View v)
             {
-                pairedDevicesList();
-            }
-        });
-
+                if (mDeviceList.getVisibility() != View.VISIBLE) {
+                    pairedDevicesList();
+
+                } else {
+                    if (mBTManager != null) {
+                        mBTManager.cancel();
+                        setConnectButtons(false);
+                    }
+                    mDeviceList.setVisibility(View.GONE);
 
+                }
+                ;
+            }
+        });
 
         //we don't want see the keyboard emulator on start
         getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN);
@@ -118,12 +155,11 @@
 
     private void pairedDevicesList() {
         mDeviceList.setVisibility(View.VISIBLE);
-
         ArrayList<String> list = mBTManager.getPairedDeviceInfos();
 
-        if (list.size()>0) {
+        if (list.size() > 0) {
             final ArrayAdapter adapter;
-            adapter = new ArrayAdapter(this,android.R.layout.simple_list_item_1, list);
+            adapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, list);
             mDeviceList.setAdapter(adapter);
             mDeviceList.setOnItemClickListener(myListClickListener); //Method called when the device from the list is clicked
         } else {
@@ -131,34 +167,23 @@
         }
     }
 
-    //reaction after selection of an BT-Device to connect
-    private AdapterView.OnItemClickListener myListClickListener = new AdapterView.OnItemClickListener()
-    {
-        public void onItemClick (AdapterView av, View v, int arg2, long arg3)
-        {
-            // Get the device MAC address, the last 17 chars in the View
-            String info = ((TextView) v).getText().toString();
-            final String address = info.substring(info.length() - 17);
-
-            mBluetoothStatus.setText(R.string.connecting);
-            mDeviceList.setVisibility(View.INVISIBLE);
-            mBTManager.connect(address);
-        }
-    };
-
     public void setConnectButtons(boolean isConnected){
             mConnect.setChecked(isConnected);
-            mConnect.setText(R.string.connect);
 
             if (!isConnected) {
                 mBluetoothStatus.setText("-");
                 mAnzParcels=0L;
                 mTextAnzParcels.setText(mAnzParcels + "");
-                mConnect.setText(R.string.disconnected);
+                mConnect.setText(R.string.connect);
             }
     }
 
 
-
+    public void changeScreen(View view) {
+        Intent intent = new Intent(this, Main2Activity.class);
+        String message = "0000";
+        intent.putExtra(EXTRA_MESSAGE, message);
+        startActivity(intent);
+    }
     }
 
Index: app/src/main/res/layout/activity_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main.xml	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/res/layout/activity_main.xml	(date 1559574375235)
@@ -87,6 +87,7 @@
     </GridLayout>
 
     <LinearLayout
+        android:id="@+id/linearLayout"
         android:layout_width="379dp"
         android:layout_height="428dp"
         android:layout_marginStart="16dp"
@@ -155,8 +156,9 @@
                 android:layout_weight="1"
                 android:checked="false"
                 android:switchPadding="10dp"
-                android:text="is Connected"
-                android:textAlignment="viewStart" />
+                android:text="@string/connect"
+                android:textAlignment="viewStart"
+                android:textOff="@string/disconnected" />
         </LinearLayout>
 
         <LinearLayout
@@ -176,4 +178,18 @@
 
     </LinearLayout>
 
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/floatingActionButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="28dp"
+        android:layout_marginBottom="16dp"
+        android:baselineAlignBottom="true"
+        android:clickable="true"
+        android:onClick="changeScreen"
+        app:backgroundTint="#F44336"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:srcCompat="@drawable/ic_tempo" />
+
 </android.support.constraint.ConstraintLayout>
\ No newline at end of file
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/res/values/strings.xml	(date 1559573655298)
@@ -1,7 +1,7 @@
 <resources>
     <string name="app_name">MotorManager</string>
     <string name="textview">TextView</string>
-    <string name="connect">"Connected"</string>
+    <string name="connect">"Show Device List"</string>
     <string name="disconnected">"Disconnected"</string>
     <string name="connecting">"Connecting..."</string>
     <string name="conn_fail">"Connection failed"</string>
Index: app/src/main/java/de/blitzkrieg/motormanager/Main2Activity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/de/blitzkrieg/motormanager/Main2Activity.java	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/java/de/blitzkrieg/motormanager/Main2Activity.java	(date 1559568992018)
@@ -1,10 +1,13 @@
 package de.blitzkrieg.motormanager;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 import android.widget.EditText;
 import android.widget.ImageView;
 
+import static de.blitzkrieg.motormanager.MainActivity.EXTRA_MESSAGE;
+
 public class Main2Activity extends AppCompatActivity {
 
     private EditText mTempo;
@@ -16,7 +19,20 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main2);
 
+        mTempo = findViewById(R.id.editText3);
+        mSign = findViewById(R.id.oneImView);
+
+        Intent intent = getIntent();
+        String msg = intent.getStringExtra(EXTRA_MESSAGE);
+        if (EXTRA_MESSAGE != null) {
+            mTempo.setText(msg);
+        } else {
+            mTempo.setText("0.0");
+        }
+
+
 
 
     }
+
 }
Index: app/src/main/AndroidManifest.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/AndroidManifest.xml	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/AndroidManifest.xml	(date 1559574581684)
@@ -16,7 +16,14 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".Main2Activity"></activity>
+        <activity
+            android:name=".Main2Activity"
+            android:parentActivityName=".MainActivity">
+            <!-- The meta-data tag is required if you support API level 15 and lower -->
+            <meta-data
+                android:name="android.support.PARENT_ACTIVITY"
+                android:value=".MainActivity" />
+        </activity>
         <activity android:name=".MainActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
Index: app/src/main/res/layout/activity_main2.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_main2.xml	(revision 563fd5af99e2cb12e5e24c5de3d5eaef930f0afa)
+++ app/src/main/res/layout/activity_main2.xml	(date 1559574104255)
@@ -13,15 +13,17 @@
         android:layout_marginStart="98dp"
         android:layout_marginTop="66dp"
         android:layout_marginEnd="32dp"
+        android:clickable="false"
         android:ems="10"
+        android:enabled="false"
         android:inputType="number"
         app:layout_constraintEnd_toStartOf="@+id/imageView2"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/imageView"
+        app:layout_constraintTop_toBottomOf="@+id/oneImView"
         tools:text="@tools:sample/us_phones" />
 
     <ImageView
-        android:id="@+id/imageView"
+        android:id="@+id/oneImView"
         android:layout_width="144dp"
         android:layout_height="129dp"
         android:layout_marginStart="133dp"
@@ -45,6 +47,6 @@
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toEndOf="@+id/editText3"
-        app:layout_constraintTop_toBottomOf="@+id/imageView"
+        app:layout_constraintTop_toBottomOf="@+id/oneImView"
         app:srcCompat="@drawable/text_kmh" />
 </android.support.constraint.ConstraintLayout>
\ No newline at end of file
